graph_api/node/node_service.py

```diff
@@ -26,21 +28,22 @@ def save_node(self, node):
         Returns:
             Result of request as node object
         """
-        create_statement = "CREATE (n) RETURN n"
-        for label in node.labels:
-            create_statement = create_statement[:-10] + f":{label}" + create_statement[-10:]
+        create_template = string.Template("CREATE (n:$labels) RETURN n")
+        create_statement = create_template.substitute(
+            labels=":".join(list(node.labels))
+        )
 
         commit_body = {
             "statements": [{"statement": create_statement}]
         }
 
-        result = requests.post(url=self.database_url,
-                               json=commit_body,
-                               auth=self.database_auth).json()
-        if len(result["errors"]) > 0:
-            result = NodeOut(errors=result["errors"])
+        response = requests.post(url=self.database_url,
+                                 json=commit_body,
+                                 auth=self.database_auth).json()
+        if len(response["errors"]) > 0:
+            result = NodeOut(errors=response["errors"])
```

Comment:
It's never a good idea to mutate the object or change the value of variable based on itself. Here we had one `result` that was the response from Neo4J request, and another `result` that was a returned value from function. They're not the same, thus it's better to have different names - i.e. `result` and `response`.